snap-core/src/main/java/org/esa/snap/core/util/ImageUtils.java

```diff
@@ -113,6 +114,21 @@ public static Dimension computeSceneRasterSize(int defaultSceneRasterWidth, int
         return new Dimension(defaultSceneRasterWidth, defaultSceneRasterHeight);
     }
 
+    public static Dimension computePreferredMosaicTileSize(int imageWidth, int imageHeight, int granularity) {
```

Comment:
I think this method is better located in the GeoTiffProductReader where it is used.
The max number of tiles is also a limitation. I think if a Geotiff has size of 160000x80000 the 10 tiles in each direction are not a good choice.
And why is it named MosaicTileSize. 
Why is the GeoTiffImageReader.computePreferredTiling not good enough anymore?

Reply:
Hi Marco,
The method you mentioned has been placed in the ImageUtils class because I want to invoke it in the readers from s2tbx project which use MultiLevelSource and JAI MosaicOp to show the band data in the UI.

The limitation is necessary because the JAI MosaicOp throws OutOfMemory if the geo tiff product size is to big (there is a geo tif test file with 129600x64800 size) and the preferred tile size is for example 256x256.

There was an OutOfMemory exception in a geo tiff test method from the acceptance tests from the snap-engine project.

Setting the ROI parameter in the JAI MosaicOP the operator throws OutOfMemory.
Therefore I use a limitation to compute the tile count in the GeoTiff MultiLevelSource.

I added the suffix MosaicTileSize  to the method name because it is used in the MultiLevelSource to create the kosaic imiage.
To avoid creation of too many tiles I incresead the tile size  for JAI MosaicOp